src/LanguageServer/Protocol/ExternalAccess/Razor/RazorComponentInfoHandler.cs

```diff
@@ -0,0 +1,133 @@
+    {
+        var project = context.Solution?.GetProject(request.Project);
+
+        if (project is null)
+        {
+            return null;
+        }
+
+        var document = context.Solution.GetDocument(request.Document);
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+        var syntaxTree = semanticModel.SyntaxTree;
+        var root = (CompilationUnitSyntax)syntaxTree.GetRoot(cancellationToken);
+
+        var (methods, fields) = GetInfoInsideRazorDocument(root, semanticModel, cancellationToken);
+
+        var result = new RazorComponentInfo
+        {
+            Methods = methods,
+            Fields = fields
+        };
+
+        return result;
+    }
+
+    private static (List<MethodInsideRazorElementInfo>, List<SymbolInsideRazorElementInfo>) GetInfoInsideRazorDocument(CompilationUnitSyntax root, SemanticModel semanticModel, CancellationToken cancellationToken)
+    {
+        var invocationExpressions = root.DescendantNodes().OfType<InvocationExpressionSyntax>().ToList();
+        var identifierNames = root.DescendantNodes().OfType<IdentifierNameSyntax>().ToList();
+        
+        List<MethodInsideRazorElementInfo> methods = [];
+        List<SymbolInsideRazorElementInfo> fields = [];
+        
+        foreach (var invocation in invocationExpressions)
+        {
+            var invocationOperation = semanticModel.GetOperation(invocation, cancellationToken) as IInvocationOperation;
+            var invocationDataFlow = semanticModel.AnalyzeDataFlow(invocation);
+            if (invocationOperation is null)
+            {
+                continue;
+            }
+
+            var targetMethod = invocationOperation.TargetMethod;
+            if (targetMethod is null)
+            {
+                continue;
+            }
+
+            var operationReturnType = invocationOperation.Type;
+            if (operationReturnType is null)
+            {
+                continue;
+            }
+
+            var parameterTypes = targetMethod.GetParameters().Select(parameter => parameter.Type.ToDisplayString()).ToList();
+            if (parameterTypes is null)
+            {
+                continue;
+            }
+
+            var methodInfo = new MethodInsideRazorElementInfo
+            {
+                Name = targetMethod.Name,
+                ReturnType = operationReturnType.ToNameDisplayString(),
+                ParameterTypes = parameterTypes
+            };
+
+            methods.Add(methodInfo);
+        }
+
+        foreach (var identifier in identifierNames)
+        {
+            var symbolInfo = semanticModel.GetSymbolInfo(identifier, cancellationToken);
+            if (symbolInfo.Symbol is IFieldSymbol or IPropertySymbol)
+            {
+                var field = new SymbolInsideRazorElementInfo
+                {
+                    Name = symbolInfo.Symbol.Name,
+                    Type = symbolInfo.Symbol.GetType().ToString()
```

Comment:
Pretty sure this is going to end up being `Microsoft.CodeAnalysis.SourceFieldSymbol` or something. Unlikely to be what you intended.